Fix the interpretations of `code_hash`

Flouse opened the pull request:
Depending on the value of `hash_type`, `code_hash` has different interpretations

Flouse committed:
Fix the interpretations of `code_hash`

Depending on the value of `hash_type`, `code_hash` has different interpretations

RetricSu commented on src/content/courses/basic-theory/chapter_5.mdx:
```diff
@@ -8,26 +8,30 @@ Lock is a script structure that looks like this:
-When unlocking a cell, we simply import the dep cell, and CKB will match the hash of the data in the dep cell with code_hash to find the corresponding code.
+Depending on the value of `hash_type`, `code_hash` has different interpretations:
+- If hash_type is "data" or "data1", code_hash should match the blake2b hash of data in a dep cell;
+- If hash_type is "type", code_hash should instead match the type script hash of a dep cell.
```
can a newbie understand `the type script hash of a dep cell`?

Flouse replied:
That's a question.
Maybe we should add some explanation for `the type script hash of a dep cell` or a link to this topic.

The example `SECP256K1_BLAKE160`'s hash_type is `type`. 
To avoid misunderstandings, I think it's better to explain this topic before showing the example.

any suggestion?

RetricSu replied:
yea it is worth explaining the concept or giving it a link.

RetricSu replied:
maybe we can add examples to these two hash_type, like a very simple graph to point to the data or whole type script part of a cell, combine with a box saying hash function, and then have some output box to connect to hash_type result

RetricSu replied:
some process flow in graph to simple display the hashing process of hash_type

Flouse replied:
It seems that all the common lock/type scripts are using `hash_type=type`.

see https://pudge.explorer.nervos.org/scripts

Flouse commented:
![image](https://user-images.githubusercontent.com/1297478/228566683-18de0113-3bcd-4ab6-8000-0ab385e7a8e6.png)

Flouse committed:
Explain the `The code locating workflow`

Locating the code of SECP256K1_BLAKE160 is a bit complicated for a
beginner developer, because it uses dep_group.
So the example was changed to finding the real code of SUDT type script.

Flouse commented:
Locating the code of SECP256K1_BLAKE160 is a bit complicated for a
beginner developer, because it uses dep_group.
So the example was changed to finding the real code of SUDT type script.

cc @RetricSu 

## Preview Link
https://ckbacademy-git-fix-codehash-interpretations-flouse.vercel.app/courses/basic-theory

### Tip
https://vercel.com/docs/concepts/deployments/comments

RetricSu reviewed (approved):
LGTM
